Migrating to sdk .net core 2.0 + VisualStudio 2017

vladimir-kucher opened the pull request:
Seems like build script is failed due to lack of project.json in VS2017 project structure

vladimir-kucher committed:
Migrating to sdk .net core 2.0 + VisualStudio 2017

vladimir-kucher committed:
trying to fix build script

vladimir-kucher committed:
Revert "trying to fix build script"

This reverts commit 52f5326af13f7ea4ad025163cd1033a6b1fa2732.

vladimir-kucher committed:
commented patch-version.sh since we don't have project.json anymore

vladimir-kucher committed:
removing samples from build

razzeee commented:
Can you check the travis config? This seems fishy to me as you probably want to use dotnetcore 2.0 here?
https://github.com/vladimir-kucher/AspNetCore.Identity.MongoDB/blob/Core2Migration/.travis.yml#L21

vladimir-kucher committed:
Replacing dotnet-dev-1.0.0-preview2.1-003177 with dotnet-dev-2.1.0-preview1-006330 in.travis.yml

vladimir-kucher committed:
setting dotnet-dev-2.1.0-preview1-006330 in global.json

razzeee commented on .travis.yml:
```diff
@@ -18,7 +18,7 @@ install:
 - echo "deb [arch=amd64] http://apt-mo.trafficmanager.net/repos/dotnet/ trusty main" | sudo tee -a /etc/apt/sources.list.d/dotnetdev.list
 - sudo apt-key adv --keyserver apt-mo.trafficmanager.net --recv-keys 417A0893 && sudo apt-get update
-- sudo apt-get install -y --no-install-recommends dotnet-dev-1.0.0-preview2.1-003177
+- sudo apt-get install -y --no-install-recommends dotnet-dev-2.1.0-preview1-006330
```
Do we need the preview? Why not `sudo apt-get install -y --no-install-recommends dotnet-sdk-2.0.0` and reverting `global.json`

vladimir-kucher replied:
Thanks. Just found proper sdk version.

vladimir-kucher committed:
changing to dotnet-sdk-2.0.0.

vladimir-kucher committed:
changing cdk lib to dotnet-sdk-2.0.0-preview3-006729

vladimir-kucher committed:
fixing build.sh

vladimir-kucher committed:
separating build of src and tests

vladimir-kucher committed:
changing sdk version: 2.0.0-preview3-25510-01

vladimir-kucher committed:
changing .yml & global.json dotnet versions to dotnet-sdk-2.0.0-preview3-006729

vladimir-kucher committed:
chnaging version in global.json

tugberkugurlu commented:
This is amazing @vladimir-kucher and @Razzeee! Thanks for doing this, will have a look now and try to get the build passing!

tugberkugurlu committed:
make that 2.0.0

tugberkugurlu committed:
only move the SDK, let's keep targetting Identity 1

Let's figure out whether Identity 1 store impl can be used on Identity 2. If that's the case, it's best to target the lowest version

tugberkugurlu committed:
fix the build script

tugberkugurlu committed:
fix travis.yml

tugberkugurlu committed:
delete that

tugberkugurlu committed:
that should be push-to-nuget.sh

tugberkugurlu commented:
This is mostly positive and lots of good things under this! A few things that worth noting:

 - Do you know whether ASP.NET Identity 1.1.2 store interface is compatible with 2.0.0? If that's the case, targeting that would be more beneficial. In any case, let's keep this PR focused to SDK move which is an amazing step on its own.

 - I assume that targeting Identity 2.0.0 package was not your main intention with this PR. Am I correct to assume that?

 - Build script needed a change the way it builds and created packages. I created https://github.com/tugberkugurlu/AspNetCore.Identity.MongoDB/tree/dotnetcore2-sdk branch on top of this. Let's observe the build of that and if it passed, I pushed my changes to your branch and merge that in! Keep in mind that this will not erase your commits and you will be seen as a contributor 🎉 

There are a few other things I have change under [dotnetcore2-sdk](https://github.com/tugberkugurlu/AspNetCore.Identity.MongoDB/tree/dotnetcore2-sdk) branch. I will point some of the important ones out as code comments.

tugberkugurlu committed:
change installation script

tugberkugurlu commented on src/AspNetCore.Identity.MongoDB/AspNetCore.Identity.MongoDB.csproj:
```diff
@@ -0,0 +1,12 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Since this is a library project, this should target `netstandard` rather than `netcoreapp`. I made that change under my branch.

tugberkugurlu commented on scripts/patch-version.sh:
```diff
@@ -1,32 +1,32 @@
-#!/bin/bash
```
We can completely get rid of this file now, it has not purpose anymore.

tugberkugurlu replied:
I have made this change on the branch as well.

tugberkugurlu committed:
verbose output

tugberkugurlu committed:
enable -internaldiagnostics

tugberkugurlu commented on .travis.yml:
```diff
@@ -18,7 +18,7 @@ install:
 - echo "deb [arch=amd64] http://apt-mo.trafficmanager.net/repos/dotnet/ trusty main" | sudo tee -a /etc/apt/sources.list.d/dotnetdev.list
 - sudo apt-key adv --keyserver apt-mo.trafficmanager.net --recv-keys 417A0893 && sudo apt-get update
-- sudo apt-get install -y --no-install-recommends dotnet-dev-1.0.0-preview2.1-003177
+- sudo apt-get install -y --no-install-recommends dotnet-sdk-2.0.0-preview3-006729
```
`dotnet-sdk-2.0.0` is what we want here. I changed this on the branch based on https://www.microsoft.com/net/core#linuxubuntu

tugberkugurlu committed:
let's try w/o nobuild

tugberkugurlu commented:
Had a few issues with running the tests under Travis CI but it all seems to be working now 🎉 once it is green, we can get this in :shipit:

tugberkugurlu commented:
It turned out that you cannot target 1.1.2 and work for 2.0.0 of identity. I made the move on https://github.com/tugberkugurlu/AspNetCore.Identity.MongoDB/pull/50
